tests/test_auto/conftest.py

```diff
@@ -35,3 +38,63 @@ class Trial(NamedTuple):
     fail_trials = [Trial(user_attrs={}, state=TrialState.FAIL)]
 
     return complete_trials + complete_trials[:3] + fail_trials
+
+
+@pytest.fixture
+def ts_with_fold_missing_tail(random_seed) -> TSDataset:
+    periods = 100
+    df1 = pd.DataFrame({"timestamp": pd.date_range("2020-01-01", periods=periods)})
+    df1["segment"] = "segment_1"
+    df1["target"] = np.random.uniform(10, 20, size=periods)
+    df1.loc[df1.index[-7:], "target"] = np.NaN
+
+    df2 = pd.DataFrame({"timestamp": pd.date_range("2020-01-01", periods=periods)})
+    df2["segment"] = "segment_2"
+    df2["target"] = np.random.uniform(-15, 5, size=periods)
+    df2.loc[df2.index[-7:], "target"] = np.NaN
+
+    df = pd.concat([df1, df2]).reset_index(drop=True)
+    df = TSDataset.to_dataset(df)
+    tsds = TSDataset(df, freq="D")
+
+    return tsds
+
+
+@pytest.fixture
+def ts_with_fold_missing_middle(random_seed) -> TSDataset:
+    periods = 100
+    df1 = pd.DataFrame({"timestamp": pd.date_range("2020-01-01", periods=periods)})
+    df1["segment"] = "segment_1"
+    df1["target"] = np.random.uniform(10, 20, size=periods)
+    df1.loc[df1.index[-14:-7], "target"] = np.NaN
+
+    df2 = pd.DataFrame({"timestamp": pd.date_range("2020-01-01", periods=periods)})
+    df2["segment"] = "segment_2"
+    df2["target"] = np.random.uniform(-15, 5, size=periods)
+    df2.loc[df2.index[-14:-7], "target"] = np.NaN
+
+    df = pd.concat([df1, df2]).reset_index(drop=True)
+    df = TSDataset.to_dataset(df)
+    tsds = TSDataset(df, freq="D")
+
+    return tsds
+
+
+@pytest.fixture
+def ts_with_few_missing(random_seed) -> TSDataset:
```

Comment:
We also need with full missing horizon

Reply:
Two previous fixtures have 7 elements missing, it was enough to make a whole fold to become missing.